index.js

```diff
@@ -154,7 +155,7 @@ function trimID (target) {
     var gitInfo = hostedGitInfo.fromUrl(target);
     if (gitInfo) {
         target = gitInfo.project;
-    } else if (isUrl(target)) {
+    } else if (isUrl(target) || isGitUrl(target)) {
         // strip away .git and everything that follows
         var strippedTarget = target.split('.git');
```

Comment:
Though this was not introduced from any of these changes in the PR. I just happened to have noticed this when reading the diffs and wanted to point it out.

If for any reason the target was `https://scm.git.foobar.com/secret/repo.git`, it would match on `scm` and not `repo`. If written in SSH format `git@scm.git.foobar.com:secret/repo.git`, it would error out.

A real example would have been cloning from a gists.
`https://gist.github.com/sample.git` or `git@gist.github.com:sample.git` but luckily this URL format is covered by the first if condition and is not a problem.

```
var gitInfo = hostedGitInfo.fromUrl(target);
if (gitInfo) {
``` 

Anyways, just pointing out something I noticed when looking over the line. Most users use anyways GitHub, Gitlab, and BitBucket which the first condition covers. +Gist

Reply:
Given that this code has entirely vanished in the master branch, I would not bother to fix this.

Reply:
(Perfect for a new issue)

Reply:
I should have looked at master before writing and re-writing my paragraph HA.

Reply:
@erisu I think the perfect sample finally came to me after working all day on unit testing it: it would be possible for a user to do `cordova plugin add https://scm.git.foobar.com/secret/repo.git#1.0.0`, expecting to get `1.0.0` release tag, but end up with the default branch due to the bug you found in this PR. Considering that this change is both broken and gone in `master` I would say that it should either be reverted or removed by rebase. I will deal with this another day.

Reply:
@brodybits I disagree. This patch does not introduce new bugs, they already existed. Why shouldn't we release it?

Reply:
@erisu you are right that the bug did exist before the change, but I think it was only in case of URL *not* starting with `git://`. The change here will now trigger the bug in case of a URL starting with `git://`. While this would be an extremely rare bug I would rather not introduce another case of such a bug in a patch release if we can avoid it somehow.

Reply:
@brodybits I tried running this sample `git://scm.service.io/user/my-repo.git` through the method and I still got back the repo name `my-repo` as expected. 

The only case where I saw this rare issue to occur was when the URL had two or more `.git`. 

In the following lines `parts = strippedTarget[0].match(re);`, it expected that index 0 of strippedTarget to contain the entire URL minus the `.git`.

Example:  `https://scm.service.io/user/my-repo`

It would then grab the item after the last `/`. 

In the rare faile case, if the URL was `https://scm.git.service.io/user/my-repo.git`, it would slit on all `.git` and in this case, index 0  becomes `https://scm`. 

It believes the repo name is `scm`.

Reply:
(Make sure to transform those into test cases for the test suite)

Reply:
I don't get why we are still discussing this. The patch fixes some use cases but breaks no previously working ones.